modules/social_features/social_event/modules/social_event_max_enroll/src/Controller/UserEnrollInviteControllerAlter.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+namespace Drupal\social_event_max_enroll\Controller;
+
+use Drupal\Core\Session\AccountProxyInterface;
+use Drupal\Core\Url;
+use Drupal\node\NodeInterface;
+use Drupal\social_event\EventEnrollmentInterface;
+use Drupal\social_event_invite\Controller\UserEnrollInviteController;
+use Drupal\social_event_max_enroll\Service\EventMaxEnrollService;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\RequestStack;
+
+/**
+ * Accepts or declines an event enrollment invite.
+ *
+ * @package Drupal\social_event_max_enroll\Controller
+ */
+class UserEnrollInviteControllerAlter extends UserEnrollInviteController {
+
+  /**
+   * The event maximum enroll service.
+   *
+   * @var \Drupal\social_event_max_enroll\Service\EventMaxEnrollService
+   */
+  protected EventMaxEnrollService $eventMaxEnrollService;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function __construct(
+    RequestStack $requestStack,
+    AccountProxyInterface $currentUser,
+    EventMaxEnrollService $eventMaxEnrollService
+  ) {
+    parent::__construct($requestStack, $currentUser);
+
+    $this->eventMaxEnrollService = $eventMaxEnrollService;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container): self {
+    return new static(
+      $container->get('request_stack'),
+      $container->get('current_user'),
+      $container->get('social_event_max_enroll.service')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function updateEnrollmentInvite(EventEnrollmentInterface $event_enrollment, string $accept_decline): RedirectResponse {
+    // We should move forward only when the user is accepting the invite.
+    if ($accept_decline === '1') {
+      // Retrieve event ID.
+      $event_id = $event_enrollment->field_event->target_id;
+
+      /** @var \Drupal\node\NodeInterface $node */
+      $node = $this->entityTypeManager()->getStorage('node')->load($event_id);
+
+      if (
+        $node instanceof NodeInterface &&
```

Comment:
Checking if $node is not null would be more applicable here as replacement for the instance check. Loading from the node storage you; can always expect the $node to be an instance of NodeInterface or to be null.

Reply:
@bojanbogdanovic instanceOf will also check if that is NULL or not. It's like apple and oranges though :-P

Reply:
@navneet0693 I understand, both cases work, but in my opinion if looking from self descriptive code perspective: instanceof gives the suggestion a different instance could be expected; which is not the case.

Reply:
Done